python/mxnet/foo/nn/layer.py

```diff
@@ -207,6 +222,80 @@ def generic_forward(self, F, x, *args, **kwargs):
         raise NotImplementedError
 
 
+class _Binary(Layer):
```

Comment:
These are very confusing. It's doing branching and merging at the same time. What's the use case? Why not just do it in forward?